### PR TITLE
feat: add docs for external secrets

### PR DESCRIPTION
**Pull request by kneal**
This PR adds the new secret documentation for release v0.6.0.

Content updates:

* Added sample in usage showing how to use them
* Added yaml tag documentation in concepts
* Updated plugin section to reflect pipeline and secret plugin types

### CODE REVIEW

File: content/concepts/pipeline/secrets/_index.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ The following fields are used to configure the component:
 | `key`    | path to secret to fetch from storage backend | `false`  |
 | `name`   | name of secret to reference in the pipeline  | `true`   |
 | `type`   | type of secret to fetch from storage backend | `false`  |
+| `type`   | loop up a secret via running a secret plugin | `false`  |
```
Comment (wass3r) [inline content/concepts/pipeline/secrets/_index.md]: \`type\` exists twice?

File: content/plugins/pipeline/registry/artifactory.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ Source Code: https://github.com/go-vela/vela-artifactory
 
 Registry: https://hub.docker.com/r/target/vela-artifactory
 
+{{% alert color="tip" %}}
+This plugin supports environment (`PARAMETER_*`) and volume (`/vela/parameters/*`) configuration for setting parameters.
```
Comment (wass3r) [inline content/plugins/pipeline/registry/artifactory.md:14]: we say `files` a few lines down, but `volume` here. should we match them up?

File: content/plugins/pipeline/registry/downstream.md [hunk 3]
```diff
@@ -10,6 +10,12 @@ Source Code: https://github.com/go-vela/vela-downstream
 
 Registry: https://hub.docker.com/r/target/vela-downstream
 
+{{% alert color="tip" %}}
+This plugin supports environment (`PARAMETER_*`) and volume (`/vela/parameters/*`) configuration for setting parameters.
```
Comment (wass3r) [inline content/plugins/pipeline/registry/downstream.md:14]: Same comment re `volume` vs `files`

File: content/plugins/pipeline/registry/kubernetes.md [hunk 4]
```diff
@@ -10,6 +10,12 @@ Source Code: https://github.com/go-vela/vela-kubernetes
 
 Registry: https://hub.docker.com/r/target/vela-kubernetes
 
+{{% alert color="tip" %}}
+This plugin supports environment (`PARAMETER_*`) and volume (`/vela/parameters/*`) configuration for setting parameters.
```
Comment (wass3r) [inline content/plugins/pipeline/registry/kubernetes.md:14]: same comment re `volume` vs `files`

File: content/plugins/pipeline/registry/s3_cache.md [hunk 5]
```diff
@@ -10,6 +10,12 @@ Source Code: https://github.com/go-vela/vela-s3-cache
 
 Registry: https://hub.docker.com/r/target/vela-s3-cache
 
+{{% alert color="tip" %}}
+This plugin supports environment (`PARAMETER_*`) and volume (`/vela/parameters/*`) configuration for setting parameters.
```
Comment (wass3r) [inline content/plugins/pipeline/registry/s3_cache.md:14]: same comment re `volume` vs `files`

File: content/plugins/pipeline/registry/terraform.md [hunk 6]
```diff
@@ -6,6 +6,12 @@ Source Code: https://github.com/go-vela/vela-terraform
 
 Registry: https://hub.docker.com/r/target/vela-terraform
 
+{{% alert color="tip" %}}
+This plugin supports environment (`PARAMETER_*`) and volume (`/vela/parameters/*`) configuration for setting parameters.
```
Comment (wass3r) [inline content/plugins/pipeline/registry/terraform.md:10]: same comment re `volume` vs `files`

File: content/plugins/secret/registry/vault.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
++       log_level: trace        
+        items:
+          # Written to path: "/vela/secrets/docker/<key>"
+          - source: secret/vela/username
+            path: docker
+```
+
+## Secrets
+
+**NOTE: Users should refrain from configuring sensitive information in your pipeline in plain text.**
```
Comment (wass3r) [inline content/plugins/secret/registry/vault.md:76]: should we use actual note/info/warning box macros here?

File: content/plugins/secret/registry/vault.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+**NOTE: Secrets used within the secret plugin must exist as Vela secrets.**
```
Comment (wass3r) [inline content/plugins/secret/registry/vault.md:78]: should we use actual note/info/warning box macros here?

File: content/usage/getting-started/workspace.md [hunk 9]
```diff
@@ -6,30 +6,62 @@ description: >
...
 ```sh
 docker volume create build-workspace
 
-docker run --volume=build-workspace:/vela/src/github.com/go-vela/hello-world <image>
+docker run --volume=build-workspace:/vela/ <image>
+```
+
+## Cloning
+
+Vela automatically checks out the repository into a local volume that is mounted into each Docker container. This volume is generally referred to as a workspace, which defines the working directory shared by all steps in a build.
```
Comment (wass3r) [inline content/usage/getting-started/workspace.md:60]: should we reference https://go-vela.github.io/docs/usage/getting-started/workspace/ here?

File: content/usage/samples/secrets_external.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
+---
+title: "External Secrets"
+toc: true
+weight: 1
+description: >
+  Sample pipeline with external secrets
+---
+
+Sample [Yaml](https://yaml.org/spec/) configuration for a project requiring a secrets to be used within a step
```
Comment (wass3r) [inline content/usage/samples/secrets_external.md:9]:
```suggestion
Sample [Yaml](https://yaml.org/spec/) configuration for a project requiring secrets to be used within a step
```

File: content/usage/samples/secrets_external.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+## Scenario
+
+User is looking to create a pipeline that can integrate with a private Vault to inject secrets that can not be used with pushing Docker image to a registry.
```
Comment (wass3r) [inline content/usage/samples/secrets_external.md]:
```suggestion
User is looking to create a pipeline that can integrate with a private Vault to inject secrets that can not be used with pushing a Docker image to a registry.
```

File: content/usage/samples/secrets_external.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+{{% alert title="Note:" color="primary" %}}
+It is assumed you have created secret `vault_token` in the web interface or [CLI](/docs/cli/).
+{{% /alert %}}
+
+The samples show a pipeline using repository secrets. Vela contains three secret types repository, organization, and shared. For examples on organization and shared please see the [secret concepts](/docs/concepts/pipeline/steps/secrets/) documentation.
```
Comment (wass3r) [inline content/usage/samples/secrets_external.md]:
```suggestion
The samples show a pipeline using repository secrets. Vela contains three secret types: repository, organization, and shared. For examples on organization and shared, please see the [secret concepts](/docs/concepts/pipeline/steps/secrets/) documentation.
```

File: content/usage/samples/secrets_external.md [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+### Steps
+
+The following [pipeline concepts](/docs/concepts/pipeline) are being used in the pipeline below:
+
+* [Steps](/docs/concepts/pipeline/steps/)
+  * [Image](/docs/concepts/pipeline/steps/image/)
+  * [Pull](/docs/concepts/pipeline/steps/pull/)*
```
Comment (wass3r) [inline content/usage/samples/secrets_external.md]:
```suggestion
  * [Pull](/docs/concepts/pipeline/steps/pull/)
```

File: content/usage/samples/secrets_external.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+  # Implicit secret definition.
+  - name: vault_token
+
+  - origin:
+      name: private vault
+      image: target/secret-vault:latest
+      pull: true
+      secrets: [ vault_token ]
+      parameters:
+        addr: vault.company.com
```
Comment (wass3r) [inline content/usage/samples/secrets_external.md]:
```suggestion
        addr: vault.example.com
```

File: content/usage/samples/secrets_external.md [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
```
Comment (wass3r) [inline content/usage/samples/secrets_external.md]:
```suggestion
        addr: vault.example.com
```

**Review by wass3r — CHANGES_REQUESTED** [review]

File: content/concepts/pipeline/secrets/_index.md (hunk 1 above)
Comment (wass3r) [reply content/concepts/pipeline/secrets/_index.md]: and `loop` -> `look`

File: content/plugins/pipeline/registry/artifactory.md (hunk 2 above)
Comment (kneal) [reply content/plugins/pipeline/registry/artifactory.md:14]: We can stick with files. Not sure which is better. Further down is `secrets` vs `parameters`

Comment (kneal) [reply content/plugins/pipeline/registry/artifactory.md:14]: Lied I'm gonna do `volume`

**Review by wass3r — APPROVED** [review]

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Pull request merged**